models/userStatus.js

```diff
@@ -325,6 +325,52 @@ const updateStatusOnTaskCompletion = async (userId) => {
   }
 };
 
+const getUsersWithoutAssignedOrInProgressTasks = async () => {
+  const usersWithoutAssignedOrInProgressTasks = [];
+  const usersNotProcessed = [];
+  let errorCount = 0;
+  let discordActiveNonArchivedUsersQuerySnapshot;
+  try {
+    discordActiveNonArchivedUsersQuerySnapshot = await usersCollection
+      .where("roles.in_discord", "==", true)
+      .where("roles.archived", "==", false)
+      .get();
+  } catch (error) {
+    logger.error(`unable to get users ${error.message}`);
+    throw new Error("unable to get users");
+  }
+  const totalValidUsersCount = discordActiveNonArchivedUsersQuerySnapshot.size;
+  if (totalValidUsersCount) {
+    await Promise.all(
+      discordActiveNonArchivedUsersQuerySnapshot.docs.map(async (userDoc) => {
+        const assigneeId = userDoc.id;
+        try {
+          const tasksQuerySnapshot = await firestore
+            .collection("tasks")
+            .where("assignee", "==", assigneeId)
+            .where("status", "in", ["ASSIGNED", "IN_PROGRESS"])
```

Comment:
we can replace the string constants used with the `TASK_STATUS.IN_PROGRESS` & `TASK_STATUS.ASSIGNED` from constants/task.js